src/Animated.test.js

```diff
@@ -0,0 +1,177 @@
+        this.anim = animation.node(this.transX, animation.config);
+      }
+      start() {
+        this.anim.start();
+      }
+      stop() {
+        this.anim.__stopImmediately_testOnly();
+      }
+      render() {
+        return (
+          <Animated.View style={{ transform: [{ translateX: this.transX }] }} />
+        );
+      }
+    }
+    const ref = React.createRef();
+
+    const initial = ReanimatedModule.getNumberOfNodes();
+    const wrapper = renderer.create(<TestComponent ref={ref} />);
+    const before = ReanimatedModule.getNumberOfNodes();
+    ref.current.start();
+    const during = ReanimatedModule.getNumberOfNodes();
+    ref.current.stop();
+    const after = ReanimatedModule.getNumberOfNodes();
+    wrapper.unmount();
+    const final = ReanimatedModule.getNumberOfNodes();
+
+    return (
+      initial === final &&
+      after === before &&
+      during > after &&
+      initial === 0 &&
+      before === 4
+    );
+  };
+
+  it('fails if timing does not attach nodes correctly', () => {
+    expect(
+      checkIfAttachAndDetachNodesProperly({
+        node: timing,
+        name: 'timing',
+        config: {
+          duration: 5000,
+          toValue: 120,
+          easing: Easing.inOut(Easing.ease),
+        },
+      })
+    ).toBeTruthy();
+  });
+
+  it('fails if decay does not attach nodes correctly', () => {
+    expect(
+      checkIfAttachAndDetachNodesProperly({
+        node: decay,
+        name: 'decay',
+        config: {
+          deceleration: 0.997,
+        },
+      })
+    ).toBeTruthy();
+  });
+
+  it('fails if spring does not attach nodes correctly', () => {
+    expect(
+      checkIfAttachAndDetachNodesProperly({
+        node: spring,
+        name: 'spring',
+        config: {
+          toValue: 0,
+          damping: 7,
+          mass: 1,
+          stiffness: 121.6,
+          overshootClamping: false,
+          restSpeedThreshold: 0.001,
+          restDisplacementThreshold: 0.001,
+        },
+      })
+    ).toBeTruthy();
+  });
+
+  it('fails if animation related nodes are still attached after detaching of value', () => {
+    const { timing, Value } = Animated;
+    class TestComponent extends React.Component {
+      constructor(props) {
+        super(props);
+        this.transX = new Value(0);
+        const config = {
+          duration: 5000,
+          toValue: -120,
+          easing: Easing.inOut(Easing.ease),
+        };
+        this.anim = timing(this.transX, config);
+        this.anim2 = timing(this.transX, config);
+      }
+      start1() {
+        this.anim.start();
+      }
+      start2() {
+        this.anim2.start();
+      }
+      render() {
+        return (
+          <Animated.View style={{ transform: [{ translateX: this.transX }] }} />
+        );
+      }
+    }
+    const ref = React.createRef();
+    const wrapper = renderer.create(<TestComponent ref={ref} />);
+
+    ref.current.start1();
+    ref.current.start2();
```

Comment:
why two animations instead of one?

Reply:
I wish to check what happens if two anims affect one node.  Changed name of test